DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/GroupDelegatesController.cs

```diff
@@ -0,0 +1,217 @@
+﻿namespace DigitalLearningSolutions.Web.Controllers.TrackingSystem.Delegates
+{
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Enums;
+    using DigitalLearningSolutions.Data.Models.User;
+    using DigitalLearningSolutions.Data.Services;
+    using DigitalLearningSolutions.Web.Attributes;
+    using DigitalLearningSolutions.Web.Helpers;
+    using DigitalLearningSolutions.Web.Models.Enums;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.GroupDelegates;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.FeatureManagement.Mvc;
+
+    [FeatureGate(FeatureFlags.RefactoredTrackingSystem)]
+    [Authorize(Policy = CustomPolicies.UserCentreAdmin)]
+    [SetDlsSubApplication(nameof(DlsSubApplication.TrackingSystem))]
+    [SetSelectedTab(nameof(NavMenuTab.Delegates))]
+    [Route("TrackingSystem/Delegates/Groups/{groupId:int}/Delegates")]
+    public class GroupDelegatesController : Controller
+    {
+        private const string AddGroupDelegateCookieName = "AddGroupDelegateFilter";
+        private readonly CentreCustomPromptHelper centreCustomPromptHelper;
+        private readonly IGroupsService groupsService;
+        private readonly IJobGroupsService jobGroupsService;
+        private readonly IUserService userService;
+
+        public GroupDelegatesController(
+            IJobGroupsService jobGroupsService,
+            IUserService userService,
+            CentreCustomPromptHelper centreCustomPromptHelper,
+            IGroupsService groupsService
+        )
+        {
+            this.centreCustomPromptHelper = centreCustomPromptHelper;
+            this.jobGroupsService = jobGroupsService;
+            this.userService = userService;
+            this.groupsService = groupsService;
+        }
+
+        [Route("{page:int=1}")]
+        public IActionResult Index(int groupId, int page = 1)
+        {
+            var centreId = User.GetCentreId();
+            var groupName = groupsService.GetGroupName(groupId, centreId);
+
+            if (groupName == null)
+            {
+                return NotFound();
+            }

```

Comment:
groupName should be guaranteed not null once we add the ServiceFilter above.

Reply:
Done.